test/python/pulse/test_builder.py

```diff
@@ -251,6 +251,23 @@ def test_phase_offset(self):
 
         self.assertEqual(schedule, reference)
 
+    def test_phase_offset_parameter(self):
```

Comment:
Have you tried `assign_parameters`? This would check that the internal structures also hold up in this context